File: 7kyu/sum-of-minimums/sum-of-minimums.py
def sum_of_minimums(numbers):
    return sum(map(lambda num: min(num), numbers))


# print(sum_of_minimums([1, 2, 3, 4, 5]))
print(sum_of_minimums([ [7,9,8,6,2 ], [6,3,5,4,3], [5,8,7,4,5] ]))

File: 7kyu/sum-of-minimums/sum-of-minimums.js
/*
PREP:
- set findMin function to find the minimum number from array
- set total function to add all minimums up
- use map and reduce
- return the result
*/

function sumOfMinimums(arr) {
	const findMin = (array) => Math.min(...array);
	const total = (prev, next) => prev + next;
	return arr.map(findMin).reduce(total, 0);
}
